llvm_util/llvm2alive.cpp

```diff
@@ -273,13 +273,32 @@ class llvm2alive_ : public llvm::InstVisitor<llvm2alive_, unique_ptr<Instr>> {
                           !known);
     unique_ptr<Instr> ret_val;
     auto argI = fn->arg_begin(), argE = fn->arg_end();
+    const auto &alist = i.getAttributes();
+    uint64_t argidx = 0;
+
     for (auto &arg : args) {
       ParamAttrs attr;
+      // TODO: Once attributes at a call site are fully supported, we should
+      // call handleAttributes().
+
+      if (uint64_t derefb = alist.getParamDereferenceableBytes(argidx)) {
```

Comment:
why do you check for dereferenceable twice? once here and once below?

Reply:
It is because a dereferenceable tag can be at either the callsite or at a function signature.
At a call instruction:
```
BB:
  call void @f(i32* dereferenceable(4) %p)
```
At a function signature:
```
declare void @f(i32* dereferenceable(4) %ptr)
```